src/FastConv/FastConv.cpp

```diff
@@ -10,23 +11,259 @@ CFastConv::~CFastConv( void )
     reset();
 }
 
-Error_t CFastConv::init(float *pfImpulseResponse, int iLengthOfIr, int iBlockLength /*= 8192*/, ConvCompMode_t eCompMode /*= kFreqDomain*/)
+Error_t CFastConv::init(float* pfImpulseResponse, int iLengthOfIr, int iBlockLength /*= 8192*/, ConvCompMode_t eCompMode /*= kFreqDomain*/)
 {
+    //set processing mode
+    if (eCompMode!= kTimeDomain and eCompMode!= kFreqDomain)
+    {
+        std::cout<< "ConvCompMode_t has to be either kTimeDomain or kFreqDomain";
```

Comment:
This could be a useful message in an exception, but the hypothetical programmer using this as a library would likely prefer that it not print to stdout.